builtin/buffer.mbt

```diff
@@ -72,6 +72,16 @@ fn grow_if_necessary(self : Buffer, required : Int) -> Unit {
   }
 }
 
+/// Return the given buffer's length in bytes.
```

Comment:
this is encoding agnostics? How is it useful

Reply:
@bobzhang This API mirrors OCaml's [`Buffer.length`](https://ocaml.org/manual/5.2/api/Buffer.html#VALlength) which counts in bytes (i.e. `char`s in OCaml), so it is indeed encoding agnostic. Currently it's feasible via `b.to_bytes().length()`, but `.to_bytes()` introduces an extra copy.

Although in Cmarkit this API is only used to check whether the buffer is empty (namely `length b <> 0`), other use cases [do exist](https://github.com/search?q=language%3AOCaml+Buffer.length&type=code), notably getting a copy of the underlying `Bytes` towards its end via [`Buffer.sub`](https://ocaml.org/manual/5.2/api/Buffer.html#VALsub) (e.g. `sub b 42 (length b - 1 - 42)`).

Thus, I think it would still be beneficial to expose the buffer length. Of course, I can still withdraw the `.length()` API if you consider this to be unnecessary.

Reply:
we are moving buffer outside builtin to provide a dedicated buffer package mimicking ocaml's behavior cc @hackwaly